src/game/SaveLoadGame.cc

```diff
@@ -1132,6 +1138,85 @@ static void SaveMercProfiles(HWFILE const f)
 	}
 }
 
+const char * IMPSavedProfileCreateFilename(const char *nickname_cstring)
```

Comment:
All `const char*` should be replaced with `ST::string`. We do not need any of the `c_str()`, `malloc` or `free`.

`sprintf` should be replaced by `ST::format`

Reply:
Once we get rid of them all, we should just poison it via `__attribute__ ((deprecated("some message.")))` to avoid reintroduction. Works on gcc/clang, not elsewhere.

Reply:
2cb2534ed77b72c0a845c9d59b2ba8b865fd689f ST::string attempt

Reply:
That commit LGTM 👍 . Thanks!